src/api/pros.py

```diff
@@ -0,0 +1,99 @@
+from flask import Blueprint, jsonify, request, abort
+import uuid
+import datetime
+import json
+from utils import load_json, save_json
+
+pros = Blueprint('pros', __name__, url_prefix='/pros')
+
+
+@pros.route('/', methods=['GET'])
+def list_pros():
+    """
+    Fetches list of professionals
+    """
+    db = load_json('./src/api/db.json')
+    if not db:
+        abort(500)
+
+    return jsonify(db), 200
+
+
+@pros.route('/', methods=['POST'])
+def create_pro():
+    """
+    Adds a new professional to directory
+    """
+    data = request.get_json()
+    if not data:
+        abort(400)
+
+    uid = uuid.uuid1()
+    if not uid:
+        abort(500)
+    try:
+        pros = {
+            'uid': str(uid),
+            'type': data['type'],
+            'firstname': data['firstname'],
+            'lastname': data['lastname'],
+            'about': data['about'],
+            'active': datetime.datetime.utcnow().timestamp(),
+            'bio': data['bio'],
+            'email': data['email'],
+            'contact': data['contact'],
+            'whatsapp': data['whatsapp'] if 'whatsapp' in data else True
+        }
+    except KeyError as e:
+        abort(400, description="{} is missing".format(e))
+
+    db = load_json('./src/api/db.json')
+
+    if not db:
+        abort(500)
+    db[str(uid)] = pros
+
+    if save_json('./src/api/db.json', db):
```

Comment:
It's usually a good pattern to end the function on the success case and escape the errors in if blocks. So, here you would do: `if not save_json(): abort(500)` instead of the other way around.

Reply:
Also, it's better if `save_json` doesn't eat the exception. If it does and just returns _True_ or _False_, the information about why it failed is lost. It may be because there are no permissions or the file already exists or the path doesn't exist etc.

The added advantage is that if `save_json()` throws an Exception, you don't need the `if save_json()` clause here. The exception will automatically get thrown above and you can catch the generic exception just like a HTTPException and return 500 in a separate errorhandler.

Reply:
Anything that is unexpected for you should be 500. Anything that is expected should be a more specific error type. For example, you're trying to open `db.json` and that file should have been created during setup, but it wasn't created. Then it is a 500. 

In a different case, if some file was requested by the request that is not on server, it would be BAD REQUEST. For eg: User's profile image was requested when it was not uploaded. That's not an issue with the API. Both cases the error is FileNotFound but HTTP code is different.

Reply:
For now, all the errors raised while opening the `db.json` file will be handled by the calling function of `load_json()` & `save_json` as the errors raised will be `500` error and this is only because a json file is our database, later when a database is incorporated we can generalise these errors into different HTTP exceptions. Do you this makes sense for now?